mlflow/types/schema.py

```diff
@@ -1,9 +1,12 @@
-import json
+import builtins
+import datetime  # pylint: disable=unused-import
```

Comment:
```suggestion
import datetime  # pylint: disable=unused-import
```

Let's remove this

Reply:
I tried 😢 I don't know why pylint keeps giving me this error
<img width="753" alt="image" src="https://github.com/mlflow/mlflow/assets/82044803/283a961e-079f-4162-9e71-a11b9ff649f8">

Reply:
because it's unsed

Reply:
I used it for DataType.datetime's python type.
<img width="272" alt="image" src="https://github.com/mlflow/mlflow/assets/82044803/d35d01f2-054f-455d-9889-e88d53dc1445">

Reply:
Maybe because it's shadowed by `datetime`?

Reply:
Updated to `from datetime import date`

Reply:
sounds good